File: app/src/main/java/com/stupidtree/cloudliter/ui/main/MainViewModel.java
package com.stupidtree.cloudliter.ui.main;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;

import com.stupidtree.cloudliter.data.model.UserLocal;
import com.stupidtree.cloudliter.data.repository.LocalUserRepository;

public class MainViewModel extends ViewModel {


    /**
     * 仓库区
     */
    //本地用户仓库
    LocalUserRepository localUserRepository;

    public MainViewModel(){
        localUserRepository = LocalUserRepository.getInstance();
    }

    @NonNull
    public UserLocal getLocalUser() {
        return localUserRepository.getLoggedInUser();
    }
}

File: app/src/main/java/com/stupidtree/cloudliter/ui/main/MainViewModel.kt
package com.stupidtree.cloudliter.ui.main

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import com.stupidtree.cloudliter.data.model.UserLocal
import com.stupidtree.cloudliter.data.repository.LocalUserRepository

class MainViewModel(application: Application?) : AndroidViewModel(application!!) {
    /**
     * 仓库区
     */
    //本地用户仓库
    var localUserRepository: LocalUserRepository = LocalUserRepository.getInstance(application!!)
    val localUser: UserLocal
        get() = localUserRepository.getLoggedInUser()

}